web/crux/src/app/team/pipes/team.reinvite-pipe.ts

```diff
@@ -0,0 +1,32 @@
+import { Injectable, PipeTransform } from '@nestjs/common'
+import { PreconditionFailedException } from 'src/exception/errors'
+import { ReinviteUserRequest } from 'src/grpc/protobuf/proto/crux'
+import PrismaService from 'src/services/prisma.service'
+import { TEAM_INVITATION_EXPIRATION } from 'src/shared/const'
+
+@Injectable()
+export default class TeamReinviteUserValidationPipe implements PipeTransform {
+  constructor(private prisma: PrismaService) {}
+
+  async transform(value: ReinviteUserRequest) {
+    const invite = await this.prisma.userInvitation.findUniqueOrThrow({
+      where: {
+        userId_teamId: {
+          userId: value.userId,
+          teamId: value.id,
+        },
+      },
+    })
+
+    const now = new Date().getTime()
```

Comment:
More readable, what do you think? 

```
    const InvitationExpired = new Date().getTime() < invite.createdAt.getTime() + TEAM_INVITATION_EXPIRATION

    if (invite.status !== 'expired' && InvitationExpired) {
...
```

Reply:
sure i guess we can make a dedicated method for this check